drivers/t5838/t5838.h

```diff
@@ -0,0 +1,226 @@
+/** @file t5838.h
+ *
+ * @brief Header for added AAD functionality for T5838 implemented in our modified PDM driver.
+ *
+ * @par
+ * COPYRIGHT NOTICE: (c) 2023 Irnas.  All rights reserved.
+ */
+
+#ifndef T5838_H
+#define T5838_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include <zephyr/device.h>
+
+/* I will call their custom one wire thin FAKE2C because it is not really one wire */
```

Comment:
I think it is good idea we keep the FAKE2C naming, since this defines shouldn't really concern end user but can clear a lot of confusion since datasheet uses their incorrect one-wire naming for the protocol.

Reply:
Ok, agree. But then you should move them to the source file, as they are not expected to be used by the user (I forgot to mention this at the first review).